### ISSUE TITLE
Adding push notifications after InAppMessaging breaks inAppMessaging on both react and reactNative

### ISSUE THREAD

**Issue by ericowhadi** [aws-amplify/amplify-js#12913]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React, React Native

### Which UI component?

In-App Messaging

### How is your app built?

Create React App

### What browsers are you seeing the problem on?

Chrome, iOS (React Native), Android (React Native)

### Which region are you seeing the problem in?

us-east-2

### Please describe your bug.

I had installed and test successfully inAppMessaging on both my webapp and react-native apps pointing to the same backend.
Everything was fine, and end to end testing was successful.
Then I added Push notification using amplify update notifications to add FCM push notification
React native apps build fine.
But now, as soon as I hit the in-app messaging code in ether of webapp or react-native app, I am getting error message Missing application id.
Looking at the amplifyconfiguration.json, appId for InAppMessaging is present and correct on both react-native and web app.
Now, I narrowed down the problem:
The additional section in "Notifications" of amplifyconfiguration.json that was added by the addition of FCM push notification is causing the bad behavior:
If I delete this section manually, inAppMessaging start working again and does not complain about missing appId. Obviously this is not an acceptable workaround, but a lead to where the problem is located...
 

### What's the expected behaviour?

should be able to have both inAppMessaging and Push notification without problem

### Help us reproduce the bug!

add both inAppMessaging and Push notification on same project (altered appId for data privacy)

Bad   amplifconfiguration.json (It looks correct but the code breaks when using it)
```js
"Analytics": {
    "AWSPinpoint": {
      "appId": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "region": "us-east-2"
    }
  },
  "Notifications": {
    "Push": {
      "AWSPinpoint": {
        "appId": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
        "region": "us-east-2"
      }
    },
    "InAppMessaging": {
      "AWSPinpoint": {
        "appId": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
        "region": "us-east-2"
      }
    }
  },
```
Good one (except that now obviously Push notification cannot work):
```js
"Analytics": {
    "AWSPinpoint": {
      "appId": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "region": "us-east-2"
    }
  },
  "Notifications": {
    "InAppMessaging": {
      "AWSPinpoint": {
        "appId": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
        "region": "us-east-2"
      }
    }
  },
```

### Code Snippet

```typescript
// Put your code below this line.

```

**Comment by calebpollman** [comment]
@ericowhadi Thanks for raising this issue. Transferring over to Amplify JS for triage

**Comment by nadetastic** [comment]
HI @ericowhadi thank you for raising this issue. Just to get some clarification:
- Is this issue happening on both ReactNative and Web applications? 
- Is your React Native application using Expo or the React Native CLI?
- Could you share you package.json or clarify what version of `aws-amplify` you are using?

The reason I ask is that Push Notifications is only supported on applications created via React Native CLI - https://docs.amplify.aws/react-native/build-a-backend/push-notifications/set-up-push-notifications/

**Comment by nadetastic** [comment]
Thank you @ericowhadi Im working to get this reproduced/identify the root issue and will follow up soon.

**Comment by jimblanc** [comment]
The fix for this issue has been released with [Amplify JS v6.0.15](https://www.npmjs.com/package/aws-amplify/v/6.0.15). Please let us know if you're run into any other issues!

**Comment by ericowhadi** [comment]
Thanks, it now works like a charm. Thanks for the quick turn around.
